src/libraries/System.Text.RegularExpressions/gen/CS0436DiagnosticSuppressor.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace System.Text.RegularExpressions.Generator
+{
+    // TODO: Delete once https://github.com/dotnet/csharplang/issues/5528 is addressed.
+
+    // RegexGenerator injects RegexGeneratorAttribute as internal and conditional into the assembly
+    // for use by code in the assembly.  However, if one assembly using RegexGenerator has
+    // InternalsVisibleTo another assembly using RegexGenerator, a CS0436 warning will be emitted
+    // highlighting the duplication.  This suppresses the benign warning.
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CS0436DiagnosticSuppressor : DiagnosticSuppressor
+    {
+        private const string AttributeName = "System.Text.RegularExpressions.RegexGeneratorAttribute";
+        private const string SuppressedId = "CS0436";
+        private const string Id = "RGDS0001";
```

Comment:
Does this Id need to be "registered" anywhere (like we do for analyzer IDs)? Or documented at all? Does it ever show up to users?

Reply:
I suppose we expect to fix the https://github.com/dotnet/csharplang/issues/5528 issue before 7.0 at which point this suppressor won't be required any longer, so that potentially means we don't need to document it, but that is just what I'm understanding.